sonar-scanner-protocol/src/main/java/org/sonar/scanner/protocol/viewer/ScannerReportViewerApp.java

```diff
@@ -335,6 +341,33 @@ private void updateMeasures(Component component) {
     }
   }
 
+  private void updateScm(Component component) {
+    scmEditor.setText("");
+    Changesets changesets = reader.readChangesets(component.getRef());
+    if (changesets == null) {
+      return;
+    }
+    List<Integer> changesetIndexByLine = changesets.getChangesetIndexByLineList();
+    try {
+      int index = 0;
+      for (Changeset changeset : changesets.getChangesetList()) {
+        scmEditor.getDocument().insertString(scmEditor.getDocument().getEndPosition().getOffset(), Integer.toString(index) + "\n", null);
+        scmEditor.getDocument().insertString(scmEditor.getDocument().getEndPosition().getOffset(), changeset.toString() + "\n", null);
```

Comment:
The `.toString` on both of these lines looks pointless.

Does the value of `scmEditor.getDocument().getEndPosition().getOffset()` between the repeated calls?